services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -56,16 +58,23 @@ import UnliftIO.Async (pooledMapConcurrentlyN_)
 import qualified Wire.API.Team.Feature as Public
 import qualified Wire.API.Team.LegalHold as Public
 
-assertLegalHoldEnabled :: TeamId -> Galley ()
-assertLegalHoldEnabled tid = unlessM (isLegalHoldEnabled tid) $ throwM legalHoldNotEnabled
+assertLegalHoldEnabledForTeam :: TeamId -> Galley ()
+assertLegalHoldEnabledForTeam tid = unlessM (isLegalHoldEnabledForTeam tid) $ throwM legalHoldNotEnabled
 
-isLegalHoldEnabled :: TeamId -> Galley Bool
-isLegalHoldEnabled tid = do
-  statusValue <- Public.tfwoStatus <$$> TeamFeatures.getFeatureStatusNoConfig @'Public.TeamFeatureLegalHold tid
-  return $ case statusValue of
-    Just Public.TeamFeatureEnabled -> True
-    Just Public.TeamFeatureDisabled -> False
-    Nothing -> False
+isLegalHoldEnabledForTeam :: TeamId -> Galley Bool
+isLegalHoldEnabledForTeam tid = do
+  view (options . Opts.optSettings . Opts.setFeatureFlags . flagLegalHold) >>= \case
+    FeatureLegalHoldDisabledPermanently -> do
+      pure False
+    FeatureLegalHoldDisabledByDefault -> do
+      statusValue <- Public.tfwoStatus <$$> TeamFeatures.getFeatureStatusNoConfig @'Public.TeamFeatureLegalHold tid
+      return $ case statusValue of
+        Just Public.TeamFeatureEnabled -> True
+        Just Public.TeamFeatureDisabled -> False
+        Nothing -> False
+    FeatureLegalHoldWhitelistTeamsAndImplicitConsent -> do
+      view (options . Opts.optSettings . Opts.setLegalHoldTeamsWhitelist)
+        <&> maybe (False {- reasonable default, even though this is impossible -}) (tid `elem`)
```

Comment:
Maybe: impossible due to option validation happening at server startup in galley.Options

Reply:
8bcd0936d0975cfc68b9b9e5804c0692e18c4ad5